### ISSUE TITLE
Western BASIC Listings Collection

### ISSUE THREAD

**Issue by gingerbeardman** [eighttails/ProgramListOCR#3]
Western BASIC was very fragmented.

We can search for basic from the following machines:

- Spectrum (built-in)
- Commodore 64 (built-in)
- Amstrad CPC (built-in)
- BBC Micro (built-in)
- Acorn Electron (built-in)
- Atari 800 (built-in)
- Apple II (many)
- IBM PC (Q Basic)
- Windows (Visual Basic, GFA BASIC)
- Amiga (AmigaBASIC, Blitz BASIC, AMOS)
- Atari ST (ST Basic, GFA BASIC, STOS)

Each had many dialects of BASIC, so it will be tricky.

**Comment by gingerbeardman** [comment]
# GFA BASIC

This was my preferred BASIC, I used it on Atari ST from 80s into 90s, and it still has an active community on a Facebook group: https://www.facebook.com/profile.php?id=431476423606129

GFA has its own file format, but could export .LST plain text listings.

Let's begin with some things and please let me know which are most useful.

Text format listing ZIP archive at
https://sites.google.com/site/gfabasic16/
direct: https://sites.google.com/site/gfabasic16/listings.zip?attredirects=0

LST files
- https://github.com/ggnkua/Atari_ST_Sources/tree/master/GFA%20Basic
- https://github.com/ggnkua/Atari_ST_Sources/tree/master/GFA%20Basic/ftp.gfa.net/pub/gfa/gfabasic/atari
- http://www.rinneberg.de/programming/gfa.htm
- https://docs.dev-docs.org/htm/search.php?find=gfa

Syntax
http://preverdy.free.fr/preverdy/kg2j/index.html

Manual
http://freddo.chez.com/GfaBasic/GFABASIC.TXT

Scans
- Atari ST Format magazine

**Comment by gingerbeardman** [comment]
# Sinclair/Spectrum BASIC

GitHub is a great source!

- https://github.com/boriel/zxbasic/tree/master/examples (20+ listings)
- https://github.com/rodrigosantosbr/sinclair_basic_zx81 (8 listings)
- https://github.com/sambrista/spectrumBasicPrograms (2 listings)
- https://github.com/jaldhar/hivolts.bas (1 listing)
- https://github.com/rondinif/XOR-in-ZX-Spectrum-basic (1 listing)

**Comment by gingerbeardman** [comment]
# C64

- https://github.com/JaganGanesh/BASIC (5 listings)
- https://github.com/mad4j/c64-codeart (1 listing)
- https://github.com/mad4j/c64-goblin/tree/master/Goblin (1 listing)

**Comment by gingerbeardman** [comment]
# BBC BASIC

- https://github.com/owainkenwayucl/BBC-BASIC

**Comment by gingerbeardman** [comment]
# Amstrad

- https://github.com/FreeTalent-BB/cpcjs/tree/master/basic-cpc
- https://github.com/backprog/Chicago
- https://github.com/cpc6128/starking
- https://github.com/FabrizioRadica/amstrad-platform-basic

Ref
- https://github.com/dfreniche/amstradbasic-vscode

**Comment by gingerbeardman** [comment]
# Apple II

- AppleSoft BASIC
- Integer BASIC
- Apple III Business BASIC
- MD-BASIC
- ORCA/Integer BASIC
- GSoft
- EhBASIC

Ref
- https://github.com/ksherlock/BASIC.tmbundle

Listings
- https://github.com/stahlmanDesign/balloon-ride-rescue
- https://github.com/bikibird/AppleIIGames

**Comment by gingerbeardman** [comment]
# Amiga

AmigaBASIC
- https://github.com/mtolly/SolarTimes
- https://github.com/tgreaves/realm-of-magik

**Comment by eighttails** [comment]
Thanks for information.

Current N6X basic language file recognizes characters in following table.
http://p6ers.net/hashi/furoku5.html
This is too redundant for Western language.
If charset is limited to ASCII it is expected to improve accuracy.

I'm planning to offer generic BASIC language model which
* Supports ASCII charset only. (drops platform or country specific characters)
* Learns many BASIC variants in one model.
I intend it achieve high accuracy in typical BASIC statements. I won't pursue rare, uncommon, tricky phrases.a

LSTM neural network doesn't rely on strict grammar so I suppose small differences between dialects doesn't matter as long as there is enough amount of training data.

**Comment by eighttails** [comment]
@gingerbeardman 
I released version 1.5.0.
It includes English user guide and generic BAS language file.
Thanks a lot.
